pkg/geo/geomfn/line_crossing_direction_test.go

```diff
@@ -0,0 +1,454 @@
+package geomfn
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/stretchr/testify/require"
+	"github.com/twpayne/go-geom"
+)
+
+func TestLineCrossingDirection(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		line1    geom.T
+		line2    geom.T
+		expected int
+	}{
+		{
+			desc:     "Two point lines, line1 vertical,line 2 horizontal crossing toward left",
+			line1:    geom.NewLineStringFlat(geom.XY, []float64{0, 0, 0, 50}),
+			line2:    geom.NewLineStringFlat(geom.XY, []float64{25, 0, -25, 0}),
+			expected: -1,
+		},
+		{
+			desc:     "Two point lines, lines with short lenghts, line1 vertical,line 2 horizontal crossing toward left",
```

Comment:
nit: lengths spelt improperly, here and below.
also nit: space after commas, e.g. `line1 vertical, line 2 horizontal`